api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -303,4 +304,105 @@ default boolean dropTable(TableIdentifier identifier) {
    * @throws NoSuchTableException if the table does not exist
    */
   Table loadTable(TableIdentifier identifier);
+
+  /**
+   * Instantiate a builder to create a table.
+   *
+   * @param identifier a table identifier
+   * @param schema a schema
+   * @return the builder to create a table
+   */
+  default Builder newTable(TableIdentifier identifier, Schema schema) {
```

Comment:
What about `buildTable` instead? This doesn't really return a table, it returns a builder that will create either a `Table` or a `Transaction`.

We may also want to mention in Javadoc that this builder is used to either create a table or start a create or replace transaction.

Reply:
Done.